src/v/resource_mgmt/tests/memory_sampling_reclaimer_test.cc

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2023 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "resource_mgmt/memory_sampling.h"
+#include "storage/batch_cache.h"
+#include "test_utils/async.h"
+
+#include <seastar/core/memory.hh>
+#include <seastar/core/reactor.hh>
+#include <seastar/testing/thread_test_case.hh>
+#include <seastar/util/log.hh>
+
+#include <boost/test/unit_test.hpp>
+
+#include <chrono>
+#include <sstream>
+#include <string>
+
+#ifndef SEASTAR_DEFAULT_ALLOCATOR
+
+/// Test batch cache integration independently
+SEASTAR_THREAD_TEST_CASE(reclaim_notifies_memory_sampling) {
```

Comment:
Very nice to see these well-written unit tests even for a component that is a bit more complicated than a plain standalone class, more like a service.